File: packages/meal-plan-engine/src/__tests__/fixtures.ts
import type { Meal, MacroTargets } from "@caloriehero/shared-types";

// ---------------------------------------------------------------------------
// Helper to create a minimal valid Meal
// ---------------------------------------------------------------------------

function makeMeal(partial: Partial<Meal> & Pick<Meal, "id" | "name" | "category" | "nutritionalInfo">): Meal {
  return {
    description: `${partial.name} - healthy Thai meal`,
    servingSize: "350g",
    price: 120,
    allergens: [],
    dietaryTags: [],
    active: true,
    createdAt: new Date("2024-01-01"),
    updatedAt: new Date("2024-01-01"),
    ...partial,
  };
}

// ---------------------------------------------------------------------------
// Breakfast meals (8 meals)
// ---------------------------------------------------------------------------

export const breakfastMeals: Meal[] = [
  makeMeal({
    id: "b1000000-0000-0000-0000-000000000001",
    name: "Khao Tom Moo (Rice Soup with Pork)",
    category: "breakfast",
    nutritionalInfo: { calories: 320, protein: 18, carbs: 45, fat: 8 },
    allergens: ["soy"],
    dietaryTags: ["halal"],
  }),
  makeMeal({
    id: "b2000000-0000-0000-0000-000000000002",
    name: "Thai Omelette with Brown Rice",
    category: "breakfast",
    nutritionalInfo: { calories: 380, protein: 22, carbs: 40, fat: 14 },
    allergens: ["eggs"],
    dietaryTags: ["gluten_free", "halal"],
  }),
  makeMeal({
    id: "b3000000-0000-0000-0000-000000000003",
    name: "Protein Smoothie Bowl",
    category: "breakfast",
    nutritionalInfo: { calories: 420, protein: 35, carbs: 38, fat: 10 },
    allergens: ["dairy"],
    dietaryTags: ["vegetarian", "high_protein", "gluten_free"],
  }),
  makeMeal({
    id: "b4000000-0000-0000-0000-000000000004",
    name: "Avocado Toast with Eggs",
    category: "breakfast",
    nutritionalInfo: { calories: 450, protein: 20, carbs: 35, fat: 22 },
    allergens: ["eggs", "wheat"],
    dietaryTags: ["vegetarian"],
  }),
  makeMeal({
    id: "b5000000-0000-0000-0000-000000000005",
    name: "Mango Sticky Rice (Low Sugar)",
    category: "breakfast",
    nutritionalInfo: { calories: 350, protein: 8, carbs: 68, fat: 6 },
    allergens: [],
    dietaryTags: ["vegetarian", "vegan", "gluten_free"],
  }),
  makeMeal({
    id: "b6000000-0000-0000-0000-000000000006",
    name: "Keto Egg Salad",
    category: "breakfast",
    nutritionalInfo: { calories: 290, protein: 18, carbs: 4, fat: 22 },
    allergens: ["eggs"],
    dietaryTags: ["keto", "low_carb", "gluten_free"],
  }),
  makeMeal({
    id: "b7000000-0000-0000-0000-000000000007",
    name: "Greek Yogurt with Granola",
    category: "breakfast",
    nutritionalInfo: { calories: 310, protein: 16, carbs: 42, fat: 7 },
    allergens: ["dairy", "wheat"],
    dietaryTags: ["vegetarian", "high_protein"],
  }),
  makeMeal({
    id: "b8000000-0000-0000-0000-000000000008",
    name: "Tofu Scramble with Vegetables",
    category: "breakfast",
    nutritionalInfo: { calories: 280, protein: 20, carbs: 18, fat: 12 },
    allergens: ["soy"],
    dietaryTags: ["vegetarian", "vegan", "gluten_free", "dairy_free"],
  }),
];

// ---------------------------------------------------------------------------
// Lunch meals (10 meals)
// ---------------------------------------------------------------------------

export const lunchMeals: Meal[] = [
  makeMeal({
    id: "l1000000-0000-0000-0000-000000000001",
    name: "Pad Thai Gai (Chicken Pad Thai)",
    category: "lunch",
    nutritionalInfo: { calories: 520, protein: 32, carbs: 58, fat: 16 },
    allergens: ["peanuts", "soy", "fish", "eggs"],
    dietaryTags: ["halal"],
  }),
  makeMeal({
    id: "l2000000-0000-0000-0000-000000000002",
    name: "Grilled Chicken Salad",
    category: "lunch",
    nutritionalInfo: { calories: 380, protein: 40, carbs: 18, fat: 14 },
    allergens: [],
    dietaryTags: ["gluten_free", "halal", "high_protein", "dairy_free"],
  }),
  makeMeal({
    id: "l3000000-0000-0000-0000-000000000003",
    name: "Tom Yum Soup with Tofu",
    category: "lunch",
    nutritionalInfo: { calories: 250, protein: 18, carbs: 20, fat: 8 },
    allergens: ["soy"],
    dietaryTags: ["vegetarian", "vegan", "gluten_free", "dairy_free"],
  }),
  makeMeal({
    id: "l4000000-0000-0000-0000-000000000004",
    name: "Som Tum with Grilled Pork",
    category: "lunch",
    nutritionalInfo: { calories: 420, protein: 35, carbs: 25, fat: 18 },
    allergens: ["peanuts", "fish"],
    dietaryTags: ["gluten_free", "halal"],
  }),
  makeMeal({
    id: "l5000000-0000-0000-0000-000000000005",
    name: "Brown Rice Beef Bowl",
    category: "lunch",
    nutritionalInfo: { calories: 580, protein: 38, carbs: 65, fat: 15 },
    allergens: ["soy"],
    dietaryTags: ["high_protein", "halal"],
  }),
  makeMeal({
    id: "l6000000-0000-0000-0000-000000000006",
    name: "Keto Larb Moo (Pork Salad)",
    category: "lunch",
    nutritionalInfo: { calories: 330, protein: 28, carbs: 6, fat: 22 },
    allergens: ["fish"],
    dietaryTags: ["keto", "low_carb", "gluten_free"],
  }),
  makeMeal({
    id: "l7000000-0000-0000-0000-000000000007",
    name: "Vegetarian Green Curry with Rice",
    category: "lunch",
    nutritionalInfo: { calories: 490, protein: 14, carbs: 72, fat: 16 },
    allergens: ["dairy"],
    dietaryTags: ["vegetarian", "gluten_free"],
  }),
  makeMeal({
    id: "l8000000-0000-0000-0000-000000000008",
    name: "Salmon with Quinoa",
    category: "lunch",
    nutritionalInfo: { calories: 520, protein: 42, carbs: 38, fat: 18 },
    allergens: ["fish"],
    dietaryTags: ["gluten_free", "high_protein", "dairy_free"],
  }),
  makeMeal({
    id: "l9000000-0000-0000-0000-000000000009",
    name: "Tofu Basil Stir Fry with Rice",
    category: "lunch",
    nutritionalInfo: { calories: 430, protein: 20, carbs: 56, fat: 12 },
    allergens: ["soy"],
    dietaryTags: ["vegetarian", "vegan", "dairy_free"],
  }),
  makeMeal({
    id: "la000000-0000-0000-0000-000000000010",
    name: "High Protein Tuna Bowl",
    category: "lunch",
    nutritionalInfo: { calories: 440, protein: 50, carbs: 32, fat: 10 },
    allergens: ["fish"],
    dietaryTags: ["high_protein", "gluten_free", "dairy_free"],
  }),
];

// ---------------------------------------------------------------------------
// Dinner meals (8 meals)
// ---------------------------------------------------------------------------

export const dinnerMeals: Meal[] = [
  makeMeal({
    id: "d1000000-0000-0000-0000-000000000001",
    name: "Massaman Curry with Chicken",
    category: "dinner",
    nutritionalInfo: { calories: 620, protein: 38, carbs: 52, fat: 26 },
    allergens: ["peanuts", "dairy"],
    dietaryTags: ["halal", "gluten_free"],
  }),
  makeMeal({
    id: "d2000000-0000-0000-0000-000000000002",
    name: "Grilled Sea Bass with Steamed Vegetables",
    category: "dinner",
    nutritionalInfo: { calories: 420, protein: 45, carbs: 18, fat: 16 },
    allergens: ["fish"],
    dietaryTags: ["gluten_free", "high_protein", "dairy_free"],
  }),
  makeMeal({
    id: "d3000000-0000-0000-0000-000000000003",
    name: "Keto Roast Duck with Vegetables",
    category: "dinner",
    nutritionalInfo: { calories: 480, protein: 40, carbs: 8, fat: 32 },
    allergens: [],
    dietaryTags: ["keto", "low_carb", "gluten_free", "dairy_free"],
  }),
  makeMeal({
    id: "d4000000-0000-0000-0000-000000000004",
    name: "Vegan Thai Red Curry",
    category: "dinner",
    nutritionalInfo: { calories: 380, protein: 12, carbs: 58, fat: 14 },
    allergens: ["soy"],
    dietaryTags: ["vegetarian", "vegan", "gluten_free"],
  }),
  makeMeal({
    id: "d5000000-0000-0000-0000-000000000005",
    name: "Beef Bulgogi Bowl",
    category: "dinner",
    nutritionalInfo: { calories: 560, protein: 42, carbs: 48, fat: 20 },
    allergens: ["soy", "sesame"],
    dietaryTags: ["high_protein", "dairy_free"],
  }),
  makeMeal({
    id: "d6000000-0000-0000-0000-000000000006",
    name: "Chicken Tikka Masala with Brown Rice",
    category: "dinner",
    nutritionalInfo: { calories: 590, protein: 44, carbs: 55, fat: 18 },
    allergens: ["dairy"],
    dietaryTags: ["halal", "high_protein", "gluten_free"],
  }),
  makeMeal({
    id: "d7000000-0000-0000-0000-000000000007",
    name: "Tempeh Stir Fry with Noodles",
    category: "dinner",
    nutritionalInfo: { calories: 440, protein: 24, carbs: 54, fat: 14 },
    allergens: ["soy", "wheat"],
    dietaryTags: ["vegetarian", "vegan", "dairy_free"],
  }),
  makeMeal({
    id: "d8000000-0000-0000-0000-000000000008",
    name: "Lean Pork Tenderloin with Sweet Potato",
    category: "dinner",
    nutritionalInfo: { calories: 500, protein: 40, carbs: 45, fat: 14 },
    allergens: [],
    dietaryTags: ["gluten_free", "dairy_free", "high_protein", "halal"],
  }),
];

// ---------------------------------------------------------------------------
// Snack meals (6 meals)
// ---------------------------------------------------------------------------

export const snackMeals: Meal[] = [
  makeMeal({
    id: "s1000000-0000-0000-0000-000000000001",
    name: "Protein Bar (Whey)",
    category: "snack",
    nutritionalInfo: { calories: 220, protein: 20, carbs: 22, fat: 6 },
    allergens: ["dairy", "soy"],
    dietaryTags: ["high_protein", "gluten_free"],
  }),
  makeMeal({
    id: "s2000000-0000-0000-0000-000000000002",
    name: "Mixed Nuts and Dried Fruit",
    category: "snack",
    nutritionalInfo: { calories: 260, protein: 8, carbs: 24, fat: 16 },
    allergens: ["tree_nuts"],
    dietaryTags: ["vegetarian", "vegan", "gluten_free", "dairy_free"],
  }),
  makeMeal({
    id: "s3000000-0000-0000-0000-000000000003",
    name: "Keto Fat Bomb (Peanut Butter)",
    category: "snack",
    nutritionalInfo: { calories: 200, protein: 6, carbs: 4, fat: 18 },
    allergens: ["peanuts"],
    dietaryTags: ["keto", "low_carb", "vegetarian", "gluten_free", "dairy_free"],
  }),
  makeMeal({
    id: "s4000000-0000-0000-0000-000000000004",
    name: "Greek Yogurt with Berries",
    category: "snack",
    nutritionalInfo: { calories: 180, protein: 14, carbs: 20, fat: 4 },
    allergens: ["dairy"],
    dietaryTags: ["vegetarian", "gluten_free", "high_protein"],
  }),
  makeMeal({
    id: "s5000000-0000-0000-0000-000000000005",
    name: "Edamame",
    category: "snack",
    nutritionalInfo: { calories: 150, protein: 12, carbs: 14, fat: 5 },
    allergens: ["soy"],
    dietaryTags: ["vegetarian", "vegan", "gluten_free", "dairy_free"],
  }),
  makeMeal({
    id: "s6000000-0000-0000-0000-000000000006",
    name: "Rice Crackers with Hummus",
    category: "snack",
    nutritionalInfo: { calories: 240, protein: 7, carbs: 36, fat: 8 },
    allergens: ["sesame"],
    dietaryTags: ["vegetarian", "vegan", "dairy_free"],
  }),
];

// ---------------------------------------------------------------------------
// Combined catalog
// ---------------------------------------------------------------------------

export const allMeals: Meal[] = [
  ...breakfastMeals,
  ...lunchMeals,
  ...dinnerMeals,
  ...snackMeals,
];

// ---------------------------------------------------------------------------
// User macro profiles
// ---------------------------------------------------------------------------

/** Standard maintenance profile — ~2000 kcal, balanced macros */
export const maintenanceTargets: MacroTargets = {
  calories: 2000,
  protein: 150,
  carbs: 200,
  fat: 65,
};

/** Bulking profile — ~2800 kcal, high carbs and protein */
export const bulkingTargets: MacroTargets = {
  calories: 2800,
  protein: 200,
  carbs: 320,
  fat: 85,
};

/** Cutting profile — ~1500 kcal, high protein, lower carbs */
export const cuttingTargets: MacroTargets = {
  calories: 1500,
  protein: 160,
  carbs: 120,
  fat: 45,
};

/** Keto profile — ~1800 kcal, very low carbs, high fat */
export const ketoTargets: MacroTargets = {
  calories: 1800,
  protein: 130,
  carbs: 30,
  fat: 140,
};

// ---------------------------------------------------------------------------
// Large catalog fixture for performance tests
// ---------------------------------------------------------------------------

/** Generates a large catalog by repeating meals with unique IDs */
export function generateLargeCatalog(count: number): Meal[] {
  const base = allMeals;
  const result: Meal[] = [];

  for (let i = 0; i < count; i++) {
    const template = base[i % base.length]!;
    result.push({
      ...template,
      id: `${template.id.slice(0, 28)}${String(i).padStart(8, "0")}`,
      name: `${template.name} #${i}`,
      // Vary nutritional info slightly
      nutritionalInfo: {
        calories: Math.max(100, template.nutritionalInfo.calories + (i % 50) - 25),
        protein: Math.max(1, template.nutritionalInfo.protein + (i % 10) - 5),
        carbs: Math.max(0, template.nutritionalInfo.carbs + (i % 10) - 5),
        fat: Math.max(1, template.nutritionalInfo.fat + (i % 6) - 3),
      },
    });
  }

  return result;
}

File: backend/tests/engine/fixtures.py
"""Test fixtures for the meal plan engine — mirrors the TypeScript fixtures."""

from app.engine.types import MacroTargets, Meal, NutritionalInfo


def make_meal(
    id: str,
    name: str,
    category: str,
    nutritional_info: NutritionalInfo,
    allergens: list[str] | None = None,
    dietary_tags: list[str] | None = None,
) -> Meal:
    return Meal(
        id=id,
        name=name,
        description=f"{name} - healthy Thai meal",
        category=category,  # type: ignore[arg-type]
        nutritional_info=nutritional_info,
        serving_size="350g",
        price=120,
        allergens=allergens or [],
        dietary_tags=dietary_tags or [],
        active=True,
    )


# --- Breakfast meals (8) ---

breakfast_meals: list[Meal] = [
    make_meal(
        "b1000000-0000-0000-0000-000000000001",
        "Khao Tom Moo (Rice Soup with Pork)",
        "breakfast",
        NutritionalInfo(calories=320, protein=18, carbs=45, fat=8),
        allergens=["soy"],
        dietary_tags=["halal"],
    ),
    make_meal(
        "b2000000-0000-0000-0000-000000000002",
        "Thai Omelette with Brown Rice",
        "breakfast",
        NutritionalInfo(calories=380, protein=22, carbs=40, fat=14),
        allergens=["eggs"],
        dietary_tags=["gluten_free", "halal"],
    ),
    make_meal(
        "b3000000-0000-0000-0000-000000000003",
        "Protein Smoothie Bowl",
        "breakfast",
        NutritionalInfo(calories=420, protein=35, carbs=38, fat=10),
        allergens=["dairy"],
        dietary_tags=["vegetarian", "high_protein", "gluten_free"],
    ),
    make_meal(
        "b4000000-0000-0000-0000-000000000004",
        "Avocado Toast with Eggs",
        "breakfast",
        NutritionalInfo(calories=450, protein=20, carbs=35, fat=22),
        allergens=["eggs", "wheat"],
        dietary_tags=["vegetarian"],
    ),
    make_meal(
        "b5000000-0000-0000-0000-000000000005",
        "Mango Sticky Rice (Low Sugar)",
        "breakfast",
        NutritionalInfo(calories=350, protein=8, carbs=68, fat=6),
        allergens=[],
        dietary_tags=["vegetarian", "vegan", "gluten_free"],
    ),
    make_meal(
        "b6000000-0000-0000-0000-000000000006",
        "Keto Egg Salad",
        "breakfast",
        NutritionalInfo(calories=290, protein=18, carbs=4, fat=22),
        allergens=["eggs"],
        dietary_tags=["keto", "low_carb", "gluten_free"],
    ),
    make_meal(
        "b7000000-0000-0000-0000-000000000007",
        "Greek Yogurt with Granola",
        "breakfast",
        NutritionalInfo(calories=310, protein=16, carbs=42, fat=7),
        allergens=["dairy", "wheat"],
        dietary_tags=["vegetarian", "high_protein"],
    ),
    make_meal(
        "b8000000-0000-0000-0000-000000000008",
        "Tofu Scramble with Vegetables",
        "breakfast",
        NutritionalInfo(calories=280, protein=20, carbs=18, fat=12),
        allergens=["soy"],
        dietary_tags=["vegetarian", "vegan", "gluten_free", "dairy_free"],
    ),
]

# --- Lunch meals (10) ---

lunch_meals: list[Meal] = [
    make_meal(
        "l1000000-0000-0000-0000-000000000001",
        "Pad Thai Gai (Chicken Pad Thai)",
        "lunch",
        NutritionalInfo(calories=520, protein=32, carbs=58, fat=16),
        allergens=["peanuts", "soy", "fish", "eggs"],
        dietary_tags=["halal"],
    ),
    make_meal(
        "l2000000-0000-0000-0000-000000000002",
        "Grilled Chicken Salad",
        "lunch",
        NutritionalInfo(calories=380, protein=40, carbs=18, fat=14),
        allergens=[],
        dietary_tags=["gluten_free", "halal", "high_protein", "dairy_free"],
    ),
    make_meal(
        "l3000000-0000-0000-0000-000000000003",
        "Tom Yum Soup with Tofu",
        "lunch",
        NutritionalInfo(calories=250, protein=18, carbs=20, fat=8),
        allergens=["soy"],
        dietary_tags=["vegetarian", "vegan", "gluten_free", "dairy_free"],
    ),
    make_meal(
        "l4000000-0000-0000-0000-000000000004",
        "Som Tum with Grilled Pork",
        "lunch",
        NutritionalInfo(calories=420, protein=35, carbs=25, fat=18),
        allergens=["peanuts", "fish"],
        dietary_tags=["gluten_free", "halal"],
    ),
    make_meal(
        "l5000000-0000-0000-0000-000000000005",
        "Brown Rice Beef Bowl",
        "lunch",
        NutritionalInfo(calories=580, protein=38, carbs=65, fat=15),
        allergens=["soy"],
        dietary_tags=["high_protein", "halal"],
    ),
    make_meal(
        "l6000000-0000-0000-0000-000000000006",
        "Keto Larb Moo (Pork Salad)",
        "lunch",
        NutritionalInfo(calories=330, protein=28, carbs=6, fat=22),
        allergens=["fish"],
        dietary_tags=["keto", "low_carb", "gluten_free"],
    ),
    make_meal(
        "l7000000-0000-0000-0000-000000000007",
        "Vegetarian Green Curry with Rice",
        "lunch",
        NutritionalInfo(calories=490, protein=14, carbs=72, fat=16),
        allergens=["dairy"],
        dietary_tags=["vegetarian", "gluten_free"],
    ),
    make_meal(
        "l8000000-0000-0000-0000-000000000008",
        "Salmon with Quinoa",
        "lunch",
        NutritionalInfo(calories=520, protein=42, carbs=38, fat=18),
        allergens=["fish"],
        dietary_tags=["gluten_free", "high_protein", "dairy_free"],
    ),
    make_meal(
        "l9000000-0000-0000-0000-000000000009",
        "Tofu Basil Stir Fry with Rice",
        "lunch",
        NutritionalInfo(calories=430, protein=20, carbs=56, fat=12),
        allergens=["soy"],
        dietary_tags=["vegetarian", "vegan", "dairy_free"],
    ),
    make_meal(
        "la000000-0000-0000-0000-000000000010",
        "High Protein Tuna Bowl",
        "lunch",
        NutritionalInfo(calories=440, protein=50, carbs=32, fat=10),
        allergens=["fish"],
        dietary_tags=["high_protein", "gluten_free", "dairy_free"],
    ),
]

# --- Dinner meals (8) ---

dinner_meals: list[Meal] = [
    make_meal(
        "d1000000-0000-0000-0000-000000000001",
        "Massaman Curry with Chicken",
        "dinner",
        NutritionalInfo(calories=620, protein=38, carbs=52, fat=26),
        allergens=["peanuts", "dairy"],
        dietary_tags=["halal", "gluten_free"],
    ),
    make_meal(
        "d2000000-0000-0000-0000-000000000002",
        "Grilled Sea Bass with Steamed Vegetables",
        "dinner",
        NutritionalInfo(calories=420, protein=45, carbs=18, fat=16),
        allergens=["fish"],
        dietary_tags=["gluten_free", "high_protein", "dairy_free"],
    ),
    make_meal(
        "d3000000-0000-0000-0000-000000000003",
        "Keto Roast Duck with Vegetables",
        "dinner",
        NutritionalInfo(calories=480, protein=40, carbs=8, fat=32),
        allergens=[],
        dietary_tags=["keto", "low_carb", "gluten_free", "dairy_free"],
    ),
    make_meal(
        "d4000000-0000-0000-0000-000000000004",
        "Vegan Thai Red Curry",
        "dinner",
        NutritionalInfo(calories=380, protein=12, carbs=58, fat=14),
        allergens=["soy"],
        dietary_tags=["vegetarian", "vegan", "gluten_free"],
    ),
    make_meal(
        "d5000000-0000-0000-0000-000000000005",
        "Beef Bulgogi Bowl",
        "dinner",
        NutritionalInfo(calories=560, protein=42, carbs=48, fat=20),
        allergens=["soy", "sesame"],
        dietary_tags=["high_protein", "dairy_free"],
    ),
    make_meal(
        "d6000000-0000-0000-0000-000000000006",
        "Chicken Tikka Masala with Brown Rice",
        "dinner",
        NutritionalInfo(calories=590, protein=44, carbs=55, fat=18),
        allergens=["dairy"],
        dietary_tags=["halal", "high_protein", "gluten_free"],
    ),
    make_meal(
        "d7000000-0000-0000-0000-000000000007",
        "Tempeh Stir Fry with Noodles",
        "dinner",
        NutritionalInfo(calories=440, protein=24, carbs=54, fat=14),
        allergens=["soy", "wheat"],
        dietary_tags=["vegetarian", "vegan", "dairy_free"],
    ),
    make_meal(
        "d8000000-0000-0000-0000-000000000008",
        "Lean Pork Tenderloin with Sweet Potato",
        "dinner",
        NutritionalInfo(calories=500, protein=40, carbs=45, fat=14),
        allergens=[],
        dietary_tags=["gluten_free", "dairy_free", "high_protein", "halal"],
    ),
]

# --- Snack meals (6) ---

snack_meals: list[Meal] = [
    make_meal(
        "s1000000-0000-0000-0000-000000000001",
        "Protein Bar (Whey)",
        "snack",
        NutritionalInfo(calories=220, protein=20, carbs=22, fat=6),
        allergens=["dairy", "soy"],
        dietary_tags=["high_protein", "gluten_free"],
    ),
    make_meal(
        "s2000000-0000-0000-0000-000000000002",
        "Mixed Nuts and Dried Fruit",
        "snack",
        NutritionalInfo(calories=260, protein=8, carbs=24, fat=16),
        allergens=["tree_nuts"],
        dietary_tags=["vegetarian", "vegan", "gluten_free", "dairy_free"],
    ),
    make_meal(
        "s3000000-0000-0000-0000-000000000003",
        "Keto Fat Bomb (Peanut Butter)",
        "snack",
        NutritionalInfo(calories=200, protein=6, carbs=4, fat=18),
        allergens=["peanuts"],
        dietary_tags=["keto", "low_carb", "vegetarian", "gluten_free", "dairy_free"],
    ),
    make_meal(
        "s4000000-0000-0000-0000-000000000004",
        "Greek Yogurt with Berries",
        "snack",
        NutritionalInfo(calories=180, protein=14, carbs=20, fat=4),
        allergens=["dairy"],
        dietary_tags=["vegetarian", "gluten_free", "high_protein"],
    ),
    make_meal(
        "s5000000-0000-0000-0000-000000000005",
        "Edamame",
        "snack",
        NutritionalInfo(calories=150, protein=12, carbs=14, fat=5),
        allergens=["soy"],
        dietary_tags=["vegetarian", "vegan", "gluten_free", "dairy_free"],
    ),
    make_meal(
        "s6000000-0000-0000-0000-000000000006",
        "Rice Crackers with Hummus",
        "snack",
        NutritionalInfo(calories=240, protein=7, carbs=36, fat=8),
        allergens=["sesame"],
        dietary_tags=["vegetarian", "vegan", "dairy_free"],
    ),
]

# --- Combined catalog ---

all_meals: list[Meal] = breakfast_meals + lunch_meals + dinner_meals + snack_meals

# --- User macro profiles ---

maintenance_targets = MacroTargets(calories=2000, protein=150, carbs=200, fat=65)
bulking_targets = MacroTargets(calories=2800, protein=200, carbs=320, fat=85)
cutting_targets = MacroTargets(calories=1500, protein=160, carbs=120, fat=45)
keto_targets = MacroTargets(calories=1800, protein=130, carbs=30, fat=140)


def generate_large_catalog(count: int) -> list[Meal]:
    """Generate a large catalog by repeating meals with unique IDs and slight macro variations."""
    base = all_meals
    result: list[Meal] = []

    for i in range(count):
        template = base[i % len(base)]
        new_id = template.id[:28] + str(i).zfill(8)
        ni = template.nutritional_info
        result.append(
            Meal(
                id=new_id,
                name=f"{template.name} #{i}",
                description=template.description,
                category=template.category,
                nutritional_info=NutritionalInfo(
                    calories=max(100, ni.calories + (i % 50) - 25),
                    protein=max(1, ni.protein + (i % 10) - 5),
                    carbs=max(0, ni.carbs + (i % 10) - 5),
                    fat=max(1, ni.fat + (i % 6) - 3),
                ),
                serving_size=template.serving_size,
                price=template.price,
                allergens=template.allergens,
                dietary_tags=template.dietary_tags,
                active=template.active,
            )
        )

    return result
